File: 1328-break-a-palindrome/1328-break-a-palindrome.rb
# @param {String} palindrome
# @return {String}
def break_palindrome(palindrome)
    len = palindrome.length
  
      return "" if len < 2

  left = len / 2
  
  for i in 0..left-1

    if palindrome[i] != 'a'
      palindrome[i] = 'a' 
    
      return palindrome
    end
  end
  
  palindrome[len-1] = 'b'

  return palindrome
end

File: 1328-break-a-palindrome/1328-break-a-palindrome.cpp
class Solution {
public:
    string breakPalindrome(string palindrome) {
      int len =palindrome.size();

      if(len == 1) return "";
      
      int first_half = len/2;
      
      for(int i=0; i<first_half; i++){
        if(palindrome[i] != 'a'){
          palindrome[i] = 'a';
          return palindrome;
        }
      }

      palindrome[len-1] = 'b';
      
      return palindrome;
    }
};